src/libstore/daemon.cc

```diff
@@ -944,16 +944,22 @@ void processConnection(
         throw Error("the Nix client version is too old");
 
     auto tunnelLogger = new TunnelLogger(to, clientVersion);
```

Comment:
I hope this could be somehow simplified. Also this could potentially be wrong (to my knowledge only when logger is changed while the tunnelLogger here is still used. As I don't have the domain specific knowledge I don't know if this is the case.